### ISSUE TITLE
timed pie countdown bug

### ISSUE THREAD

**Issue by xRoge1x** [TheRealJoelmatic/RemoveAdblockThing#254]
**Describe the bug**

After the webpage loads the ad is skipped to the end and it looks like the "Skip ad" button is being "clicked" by the scrip but it does not activate until the "timed pie countdown" reaches zero. 

During this time the video player is stuck on the opaque last image of the ad. 

After this happens it moves on to the next ad and repeats the process. 

If I am fast enough I can click on the " Skip ad" button myself and it cuts the pie countdown short. 

I think i triggered this by trying, instinctively, to click on the "skip ad" button when there was a delay in the script clicking it for me. 

I have tried deleting and reinstalling the script but it still does it. 

Don't know if this is something you can fix on your end as the script is the current version unmodified but just thought I would let you know. 

**To Reproduce**
1. Go to 'youtube.com'
2. Click on 'any video'

**Expected behavior**
The adds to be skipped and the video to play after a brief pause to allows the script to work.

**Screenshots**
Here is a Loom video of the bug.
https://www.loom.com/share/79356fd29b0a473380573f736a9dbbd0?sid=ed65fff7-ba00-498c-8e8f-c58438b11dd8

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: Chrome
 - Version: 2.9

**Additional context**

**Comment by ToiYeuTien** [comment]
I also encountered the same situation!

**Comment by ToiYeuTien** [comment]
It seems the author has updated, very quickly, thank you for your contribution!

**Comment by xRoge1x** [comment]
It is still happening to me even after the recent update. I think it may just be an issue on my end so I will try to trouble shoot it.

**Comment by Kiiiiiwiiiii** [comment]
> It is still happening to me even after the recent update. I think it may just be an issue on my end so I will try to trouble shoot it.

Please let me know if you find a fix, the fix that the authors have put in did not fix it on my end either so I am wondering if something you do will help me as well. Thanks

**Comment by TheRealJoelmatic** [comment]
Ok, it seems interesting based on the video I think it has to do with the script not clicking the "skip" button.  As I can't seem to reproduce it right now after trying I guess the skip button is missing/not finding one of these.
```
const skipBtn = document.querySelector('.videoAdUiSkipButton,.ytp-ad-skip-button');
const nonVid = document.querySelector(".ytp-ad-skip-button-modern");
 ```
YouTube likes to be annoying and sometimes changes the names of stuff to try and block scripts like this. To fix this I just need the updated name for the button.

**Comment by xRoge1x** [comment]
Okay. I'll try to inspect it and see if they changed the name of the skip button.

**Comment by xRoge1x** [comment]
Here is a new video of it with the most recent update 3.1

https://www.loom.com/share/68ce8f76822648cf91eeefbcd1b9f3db?sid=2a964d85-9e87-4816-99e5-dd9ca5ec308e

**Comment by MahbbRah** [comment]
confirmed that this bug still persists

**Comment by TheRealJoelmatic** [comment]
> Here is a new video of it with the most recent update 3.1
> 
> https://www.loom.com/share/68ce8f76822648cf91eeefbcd1b9f3db?sid=2a964d85-9e87-4816-99e5-dd9ca5ec308e

Fixed in 3.2
